packages/neuron-wallet/src/block-sync-renderer/index.ts

```diff
@@ -73,74 +78,73 @@ export const createBlockSyncTask = async (clearIndexerFolder = false) => {
     IndexerFolderManager.resetIndexerData()
   }
 
-  if (backgroundWindow) {
+  if (syncTask) {
     return
   }
 
   logger.info('Sync:\tstarting background process')
-  backgroundWindow = new BrowserWindow({
-    width: 1366,
-    height: 768,
-    show: false,
-    webPreferences: {
-      nodeIntegration: false,
-      preload: path.join(__dirname, './preload.js')
-    }
-  })
-
-  backgroundWindow.on('ready-to-show', async () => {
-    if (!network) {
-      network = NetworksService.getInstance().getCurrent()
-    }
 
-    const startBlockNumber = (await new SyncedBlockNumber().getNextBlock()).toString()
-    SyncedBlockNumberSubject.getSubject().next(startBlockNumber)
-    logger.info('Sync:\tbackground process started, scan from block #' + startBlockNumber)
-
-    DataUpdateSubject.next({
-      dataType: 'transaction',
-      actionType: 'update',
-    })
-
-    if (network.genesisHash !== EMPTY_GENESIS_HASH) {
-      // re init txCount in addresses if switch network
-      await updateAllAddressesTxCountAndUsedByAnyoneCanPay(network.genesisHash)
-      if (backgroundWindow) {
-        const addressesMetas = AddressService.allAddresses()
-        backgroundWindow.webContents.send(
-          "block-sync:start",
-          network.remote,
-          network.genesisHash,
-          addressesMetas
-        )
+  syncTask = await spawn<SyncTask>(
+    fork(path.join(__dirname, 'task.js'), [], {
+      env: {
+        fileBasePath: env.fileBasePath
       }
+    })
+  )
+
+  subscribe(syncTask, msg => {
```

Comment:
Renaming this local function name to `subscribeToWorkerProcess` would be more expressive.